### ISSUE TITLE
localizedRead - localeFallbacks not applying to list media items

### ISSUE THREAD

**Issue by robertp-harrys** [datocms/js-datocms-client#124]
Hello!

We're trying to upgrade to latest gatsby version. During upgrade to gatsby v3. using `graphql-source-datocms` in versions both `3.0.6` we are run into the issue where localeFallbacks don't work for multiple media fields. Same issues appear in v2.6.17 (gatsby2). The last working version we're currently use is `2.4.9` for `gatsby-source-datocms`.

Below i show a screenshot of the difference in results returned in different versions of the plugin.
![fallback-query-difference](https://user-images.githubusercontent.com/61972229/139673794-79af7165-00be-42c6-b565-a906e7991347.png)

Going through the source code of `datocms-client` I've found a potential bug in `localizedRead` util.

https://github.com/datocms/js-datocms-client/blob/master/src/utils/localizedRead.js#L5-L11

when localeValue is an empty Array: 
```localeValue !== null && localeValue !== undefined && localeValue !== ''```
^ the above check will return true resulting in empty array being returned from the query.

I've locally made a change in the same line to: 

```
localeValue !== null && localeValue !== undefined && localeValue !== '' && localeValue.length !== 0
```

Which fixed the issue of missing images.

**Comment by stefanoverna** [comment]
You're right @robertp-harrys! Just released 3.5.8 which hopefully will fix the issue!
